lib/transport.ts

```diff
@@ -65,10 +61,10 @@ export class Transport {
       if (this.site !== undefined && this.flushMetricsToLogs === false) {
         lam.addEnvironment(siteURLEnvVar, this.site);
       }
-      if (this.apiKey !== undefined && this.flushMetricsToLogs === false) {
+      if (this.apiKey !== undefined) {
         lam.addEnvironment(apiKeyEnvVar, this.apiKey);
       }
-      if (this.apiKMSKey !== undefined && this.flushMetricsToLogs === false) {
+      if (this.apiKMSKey !== undefined) {
         lam.addEnvironment(apiKeyKMSEnvVar, this.apiKMSKey);
```

Comment:
These changes make sure that if an API key is provided then we will always add it as an env variable